src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -56,21 +56,24 @@ public sealed partial class JsonDocument : IDisposable
         /// <inheritdoc />
         public void Dispose()
         {
-            if (_utf8Json.IsEmpty || !IsDisposable)
+            int length = _utf8Json.Length;
+            if (length == 0 || !IsDisposable)
             {
                 return;
             }
 
-            int length = _utf8Json.Length;
-            _utf8Json = ReadOnlyMemory<byte>.Empty;
             _parsedData.Dispose();
+            _utf8Json = ReadOnlyMemory<byte>.Empty;
 
             // When "extra rented bytes exist" they contain the document,
             // and thus need to be cleared before being returned.
-            if (_extraRentedBytes != null)
+            byte[] extraRentedBytes = Threading.Interlocked.Exchange(ref _extraRentedBytes, null);
+
+            if (extraRentedBytes != null)
             {
-                _extraRentedBytes.AsSpan(0, length).Clear();
-                ArrayPool<byte>.Shared.Return(_extraRentedBytes);
+                extraRentedBytes.AsSpan(0, length).Clear();
+                ArrayPool<byte>.Shared.Return(extraRentedBytes);
+
                 _extraRentedBytes = null;
```

Comment:
Do we still need this line? Should this be `extraRentedBytes = null`?

Reply:
It seems Interlocked.Exchange already sets it to null, removing the assignment then, thanks.